load/load.py

```diff
@@ -33,7 +34,8 @@ def get_next_batch(self):
         return self.ds.batches.get()
 
     def stop_batch_creation(self):
-        self.event.set()
+        self.stop_bc.set()
+        self.stop_sc.set()
```

Comment:
Having separate events prevents lock contention for the event lock.
This does not mean that the sample_creator process ends immediately. If it has started creating a sample, it will not check for the event until the sample creation is complete.